gradio/interface.py

```diff
@@ -450,9 +450,14 @@ def clean_html(raw_html):
             cache_interface_examples(self)
 
         if self.allow_flagging != "never":
```

Comment:
you could make a variable called `all_components = self.input_components + (self.output_components if self.interface_type =...)`
and then use `all_components` everywhere instead of repeating so much flagging logic

Reply:
Yeah ok